File: codewars/python/find_the_missing_letter.py
# https://www.codewars.com/kata/5839edaa6754d6fec10000a2/train/python


def find_missing_letter(chars):
    letters = ""

    for i in chars:
        letters += i

    curr = ord(letters[0])

    for char in letters:
        if ord(char) == curr:
            curr += 1
        else:
            return chr(curr)

    return chr(curr + 1)

if __name__ == "__main__":
    #chars = ['a','b','c','d','f']
    chars = ['O','Q','R','S']

    print(find_missing_letter(chars))

File: codewars/c/find_the_missing_letter.c
// https://www.codewars.com/kata/5839edaa6754d6fec10000a2/train/c

#include <stdio.h>
#include <string.h>

char findMissingLetter(char array[], int arrayLength) {
	char ascii[arrayLength];

	for(int i = 0; i < arrayLength; i++) {
		ascii[i] = (int) array[i];
	}

	int curr = ascii[0];

	for(int i = 0; i < arrayLength; ++i) {
		if( ascii[i] == curr ) {
			curr += 1;
		} else {
			return (char) curr;
		}
	}

	return ' ';
}


int main(int argc, char* argv[]) {
	char arr1[] = { 'a', 'b', 'c', 'd', 'f' };

	printf("%c\n", findMissingLetter(arr1, sizeof(arr1)));
}
